support/cas-server-support-pm/src/main/java/org/apereo/cas/pm/BasePasswordManagementService.java

```diff
@@ -119,8 +119,9 @@ public boolean change(final Credential c, final PasswordChangeBean bean) {
      * @param c    the credential
      * @param bean the bean
      * @return the boolean
+     * @throws InvalidPasswordException if new password fails downstream validation
      */
-    public boolean changeInternal(final Credential c, final PasswordChangeBean bean) {
+    public boolean changeInternal(final Credential c, final PasswordChangeBean bean) throws InvalidPasswordException {
```

Comment:
All great. Thank you. However, do unchecked exceptions need to be part of the throws clause? (I suppose they are, if you want to add javadocs; just checking)

Reply:
I don't think so, and there's a catch for any exceptions there, but without a throws it's hard to tell that it **can** be thrown and handled properly without seeing all the use cases.

Reply:
For API documentation purposes, it wouldn't hurt.

Reply:
Sorry, were you asking me to add something? I didn't catch that. There are @throws in the javadoc for the methods, am I missing one somewhere?

Reply:
No, all good. I guess I was really thinking out loud really; I remember IntelliJ seemed to bug me about declaring runtime exceptions, but it did also complain about a `@throws` clause being there in the docs without an explicit declaration of the exception class. Catch 20-20. 

All is well. Carry on :)

Reply:
Ah. I'm an eclipse user and this is my first java work in a **very long** time. I'm noticing conventions have changed a lot. :)